source/gui/settingsDialogs.py

```diff
@@ -2050,13 +2050,36 @@ def __init__(self, parent):
 		self.UIAInMSWordCheckBox.SetValue(config.conf["UIA"]["useInMSWordWhenAvailable"])
 		self.UIAInMSWordCheckBox.defaultValue = self._getDefaultValue(["UIA", "useInMSWordWhenAvailable"])
 
-		# Translators: This is the label for a checkbox in the
-		#  Advanced settings panel.
-		label = _("Use UI Automation to access the Windows Console when available")
-		consoleUIADevMap = True if config.conf['UIA']['winConsoleImplementation'] == 'UIA' else False
-		self.ConsoleUIACheckBox=UIAGroup.addItem(wx.CheckBox(self, label=label))
-		self.ConsoleUIACheckBox.SetValue(consoleUIADevMap)
-		self.ConsoleUIACheckBox.defaultValue = self._getDefaultValue(["UIA", "winConsoleImplementation"])
+		# Translators: This is the label for a combo box for selecting the
+		# active console implementation in the advanced settings panel.
+		# Choices are automatic, prefer UIA, and force legacy.
+		consoleComboText = _("Windoes C&onsole support:")
+		consoleChoices=[
+			#Translators: A choice in a combo box in the advanced settings
+			# panel to have NVDA determine its Windows Console implementation
+			# automatically.
+			_("Automatic"),
+			#Translators: A choice in a combo box in the advanced settings
+			# panel to have NVDA use UIA in the Windows Console when available.
+			_("Prefer UIA"),
+			#Translators: A choice in a combo box in the advanced settings
+			# panel to have NVDA use its legacy Windoes Console support
+			# in all cases.
+			_("Force legacy")
+		]
+		self.consoleCombo = UIAGroup.addLabeledControl(consoleComboText, wx.Choice, choices=consoleChoices)
+		consoleValsToIndices = {
+			"auto": 0,
+			"UIA": 1,
+			"legacy": 2
+		}
+		curChoice = consoleValsToIndices[
+			config.conf['UIA']['winConsoleImplementation']
+		]
+		self.consoleCombo.SetSelection(curChoice)
+		self.consoleCombo.defaultValue = consoleValsToIndices[
+			self._getDefaultValue(["UIA", "winConsoleImplementation"])
+		]
 
 		# Translators: This is the label for a checkbox in the
 		#  Advanced settings panel.

```

Comment:
Could you make it so that this checkbox is hidden when the former combo box is set to legacy? See for an example how the open scratchpad dir button is activated when the checkbox is ticked.